freqtrade/freqtradebot.py

```diff
@@ -252,6 +252,24 @@ def get_target_bid(self, ticker: Dict[str, float]) -> float:
         balance = self.config['bid_strategy']['ask_last_balance']
         return ticker['ask'] + balance * (ticker['last'] - ticker['ask'])
 
+    def _get_trade_stake_amount(self) -> float:
+        stake_amount = self.config['stake_amount']
+        avaliable_amount = exchange.get_balance(self.config['stake_currency'])
+
+        if stake_amount == constants.UNLIMITED_STAKE_AMOUNT:
+            open_trades = len(Trade.query.filter(Trade.is_open.is_(True)).all())
+            if open_trades == self.config['max_open_trades']:
+                return 0
+            return avaliable_amount / (self.config['max_open_trades'] - open_trades)
```

Comment:
Not sure, but i think this can return negative amounts if max_open_trades is 0 (which is an allowed value).
(setting max_open_trades to 0 is used to stop trading - close open trades regularily, but don't open new ones).